File: EsalPlatform/secrets/env_loader.py
"""
Environment Configuration Loader for Python
Loads environment variables from the secrets folder based on ENVIRONMENT variable
"""

import os
from pathlib import Path
from typing import Optional, Dict, Any
from dotenv import load_dotenv


class EnvironmentConfig:
    """Environment configuration class with validation"""
    
    def __init__(self, environment: Optional[str] = None):
        self.environment = environment or os.getenv('ENVIRONMENT', 'development')
        self._load_environment()
        self._validate_required_vars()
    
    def _load_environment(self):
        """Load environment variables from secrets file"""
        secrets_path = Path(__file__).parent / 'environments' / f'.env.{self.environment}'
        
        if secrets_path.exists():
            load_dotenv(secrets_path)
        else:
            print(f"Warning: Secrets file not found: {secrets_path}")
            print("Loading from system environment variables instead...")
    
    def _validate_required_vars(self):
        """Validate that required environment variables are present"""
        required_vars = [
            'SUPABASE_URL',
            'SUPABASE_ANON_KEY',
            'JWT_SECRET_KEY',
            'DATABASE_URL'
        ]
        
        missing = [var for var in required_vars if not os.getenv(var)]
        if missing:
            raise ValueError(f"Missing required environment variables: {', '.join(missing)}")
    
    @property
    def supabase_url(self) -> str:
        return os.getenv('SUPABASE_URL', '')
    
    @property
    def supabase_anon_key(self) -> str:
        return os.getenv('SUPABASE_ANON_KEY', '')
    
    @property
    def supabase_service_role_key(self) -> str:
        return os.getenv('SUPABASE_SERVICE_ROLE_KEY', '')
    
    @property
    def database_url(self) -> str:
        return os.getenv('DATABASE_URL', '')
    
    @property
    def gemini_api_key(self) -> str:
        return os.getenv('GEMINI_API_KEY', '')
    
    @property
    def openai_api_key(self) -> Optional[str]:
        return os.getenv('OPENAI_API_KEY')
    
    @property
    def jwt_secret_key(self) -> str:
        return os.getenv('JWT_SECRET_KEY', '')
    
    @property
    def jwt_algorithm(self) -> str:
        return os.getenv('JWT_ALGORITHM', 'HS256')
    
    @property
    def jwt_expiration_time(self) -> int:
        return int(os.getenv('JWT_EXPIRATION_TIME', '3600'))
    
    @property
    def debug(self) -> bool:
        return os.getenv('DEBUG', 'false').lower() == 'true'
    
    @property
    def project_name(self) -> str:
        return os.getenv('PROJECT_NAME', 'ESAL Platform API')
    
    @property
    def api_version(self) -> str:
        return os.getenv('API_VERSION', 'v1')
    
    @property
    def allowed_origins(self) -> list:
        origins = os.getenv('ALLOWED_ORIGINS', 'http://localhost:3000')
        return [origin.strip() for origin in origins.split(',')]
    
    @property
    def smtp_host(self) -> str:
        return os.getenv('SMTP_HOST', 'smtp.gmail.com')
    
    @property
    def smtp_port(self) -> int:
        return int(os.getenv('SMTP_PORT', '587'))
    
    @property
    def smtp_user(self) -> str:
        return os.getenv('SMTP_USER', '')
    
    @property
    def smtp_password(self) -> str:
        return os.getenv('SMTP_PASSWORD', '')
    
    @property
    def smtp_from_email(self) -> str:
        return os.getenv('SMTP_FROM_EMAIL', '')
    
    @property
    def smtp_from_name(self) -> str:
        return os.getenv('SMTP_FROM_NAME', 'ESAL Platform')
    
    @property
    def redis_url(self) -> str:
        return os.getenv('REDIS_URL', 'redis://localhost:6379/0')
    
    def to_dict(self) -> Dict[str, Any]:
        """Return configuration as dictionary"""
        return {
            'environment': self.environment,
            'supabase_url': self.supabase_url,
            'database_url': self.database_url,
            'debug': self.debug,
            'project_name': self.project_name,
            'api_version': self.api_version,
            'allowed_origins': self.allowed_origins,
            'smtp_host': self.smtp_host,
            'smtp_port': self.smtp_port,
            'redis_url': self.redis_url
        }


# Create singleton instance
env_config = EnvironmentConfig()


def get_environment_config(environment: Optional[str] = None) -> EnvironmentConfig:
    """Get environment configuration instance"""
    if environment:
        return EnvironmentConfig(environment)
    return env_config

File: EsalPlatform/secrets/env-loader.ts
/**
 * Environment Configuration Loader
 * Loads environment variables from the secrets folder based on NODE_ENV
 */

import { config } from "dotenv";
import { join } from "path";
import { existsSync } from "fs";

export interface EnvironmentConfig {
  // Supabase
  SUPABASE_URL: string;
  SUPABASE_ANON_KEY: string;
  SUPABASE_SERVICE_ROLE_KEY: string;

  // Database
  DATABASE_URL: string;

  // AI Services
  GEMINI_API_KEY: string;
  OPENAI_API_KEY?: string;

  // JWT
  JWT_SECRET_KEY: string;
  JWT_ALGORITHM: string;
  JWT_EXPIRATION_TIME: string;

  // Application
  DEBUG: string;
  ENVIRONMENT: string;
  PROJECT_NAME: string;
  API_VERSION: string;

  // CORS
  ALLOWED_ORIGINS: string;

  // Email
  SMTP_HOST: string;
  SMTP_PORT: string;
  SMTP_USER: string;
  SMTP_PASSWORD: string;
  SMTP_FROM_EMAIL: string;
  SMTP_FROM_NAME: string;

  // Redis
  REDIS_URL: string;
}

/**
 * Load environment variables from the secrets folder
 * @param environment - The environment to load (development, production, test)
 * @returns Parsed environment configuration
 */
export function loadEnvironment(environment?: string): EnvironmentConfig {
  const env = environment || process.env.NODE_ENV || "development";
  const secretsPath = join(
    process.cwd(),
    "secrets",
    "environments",
    `.env.${env}`
  );

  // Check if secrets file exists
  if (!existsSync(secretsPath)) {
    console.warn(`Secrets file not found: ${secretsPath}`);
    console.warn("Loading from process.env instead...");
  } else {
    // Load from secrets file
    config({ path: secretsPath });
  }

  // Validate required environment variables
  const requiredVars = [
    "SUPABASE_URL",
    "SUPABASE_ANON_KEY",
    "JWT_SECRET_KEY",
    "DATABASE_URL",
  ];

  const missing = requiredVars.filter((key) => !process.env[key]);
  if (missing.length > 0) {
    throw new Error(
      `Missing required environment variables: ${missing.join(", ")}`
    );
  }

  return {
    SUPABASE_URL: process.env.SUPABASE_URL!,
    SUPABASE_ANON_KEY: process.env.SUPABASE_ANON_KEY!,
    SUPABASE_SERVICE_ROLE_KEY: process.env.SUPABASE_SERVICE_ROLE_KEY!,
    DATABASE_URL: process.env.DATABASE_URL!,
    GEMINI_API_KEY: process.env.GEMINI_API_KEY!,
    OPENAI_API_KEY: process.env.OPENAI_API_KEY,
    JWT_SECRET_KEY: process.env.JWT_SECRET_KEY!,
    JWT_ALGORITHM: process.env.JWT_ALGORITHM || "HS256",
    JWT_EXPIRATION_TIME: process.env.JWT_EXPIRATION_TIME || "3600",
    DEBUG: process.env.DEBUG || "false",
    ENVIRONMENT: process.env.ENVIRONMENT || env,
    PROJECT_NAME: process.env.PROJECT_NAME || "ESAL Platform API",
    API_VERSION: process.env.API_VERSION || "v1",
    ALLOWED_ORIGINS: process.env.ALLOWED_ORIGINS || "http://localhost:3000",
    SMTP_HOST: process.env.SMTP_HOST || "smtp.gmail.com",
    SMTP_PORT: process.env.SMTP_PORT || "587",
    SMTP_USER: process.env.SMTP_USER!,
    SMTP_PASSWORD: process.env.SMTP_PASSWORD!,
    SMTP_FROM_EMAIL: process.env.SMTP_FROM_EMAIL!,
    SMTP_FROM_NAME: process.env.SMTP_FROM_NAME || "ESAL Platform",
    REDIS_URL: process.env.REDIS_URL || "redis://localhost:6379/0",
  };
}

// Export singleton instance
export const env = loadEnvironment();
